x-pack/plugins/security/public/session/session_timeout.tsx

```diff
@@ -206,4 +214,28 @@ export class SessionTimeout implements ISessionTimeout {
     }
     this.warningToast = this.notifications.toasts.add(toast);
   };
+
+  /**
+   * Starts a timer that uses a native `setTimeout` under the hood. When `timeout` is larger
+   * than the maximum supported one then method calls itself recursively as many times as needed.
+   * @param updater Method that is supposed to update a reference to a native timer ID that can be
+   * used with native `clearTimeout`. It's essential for the larger timeouts when `setTimeout` is
+   * called multiple times and timer ID changes.
+   * when timer ID changes
+   * @param callback A function to be executed after the timer expires.
+   * @param timeout The time, in milliseconds the timer should wait before the specified function is
+   * executed.
+   */
+  private startTimer(updater: (timeoutID: number) => void, callback: () => void, timeout: number) {
+    // Max timeout is the largest possible 32-bit signed integer or 2,147,483,647 or 0x7fffffff.
+    const maxTimeout = 0x7fffffff;
```

Comment:
__note:__ didn't want to create a dedicated top level const for this purely "technical" thing....